modules/ndu/src/backend/training/trainer.ts

```diff
@@ -0,0 +1,103 @@
+import * as sdk from 'botpress/sdk'
+import _ from 'lodash'
+import ms from 'ms'
+
+import { broadcastLoadModel } from '..'
+import { Features, InputData } from '../typings'
+import { computeModelHash } from '../utils'
+
+import { BASE_DATA } from './base-data'
+import * as ModelService from './model-service'
+
+export const ActionTypes = <const>[
+  'faq_trigger_outside_topic',
+  'faq_trigger_inside_topic',
+  'wf_trigger_inside_topic',
+  'wf_trigger_outside_topic',
+  'wf_trigger_inside_wf',
+  'faq_trigger_inside_wf',
+  'node_trigger_inside_wf'
+]
+
+export type ActionType = typeof ActionTypes[number]
+export type ActionPredictions = { [key in ActionType]: number }
+
+const newFeature = (): Features => ({
```

Comment:
I know you simply copy/pasted this here but do you happen to know why this is called newFeature hehe ? 
Oh I think I get it now, it's a disguised `feature factory func`